CMP1903_A1_2324/Testing.cs

```diff
@@ -8,13 +9,22 @@ namespace CMP1903_A1_2324
 {
     internal class Testing
     {
-        /*
-         * This class should test the Game and the Die class.
-         * Create a Game object, call the methods and compare their output to expected output.
-         * Create a Die object and call its method.
-         * Use debug.assert() to make the comparisons and tests.
-         */
-
-        //Method
+        public void testing()
+        {
+            Game play_game = new Game(); // creates game object
+            Console.WriteLine("----------TESTING----------"); //signifies that this run is a test run to verify the numbers we expect are correct
+            int[] diceRoll_list = play_game.DiceStatsAndReport(); // calls method from Game class
+            int sum = 0;
+            for (int count = 0; count <=2; count++) // loop checks that each roll is between 1 and 6, if not then an error message will display
+            {
+                Debug.Assert(diceRoll_list[count] >= 1 && diceRoll_list[count] <=6, "The number die " + (count+1) + " rolled WASN'T between 1 and 6 so there is an error");
+                sum = sum + diceRoll_list[count];
+                if (count == 2) // checks to see if the loop is on its final iteration
+                {
+                    Debug.Assert(sum >= 6 && sum <= 18, "The sum of the rolls was NOT between 6 and 18 - Error has occured"); // checks sum to make sure it is betweem 6 and 18
+                    // if not, error message will display
+                }
+            }
+        }
     }
```

Comment:
Sufficient use of Debug.Assert() for error handling. There is a logic error where Debug.Assert() is checking that the dice roll total is between 6 and 18, where it should be 3 and 18.